pkl-core/src/main/java/org/pkl/core/stdlib/base/MappingNodes.java

```diff
@@ -50,9 +51,14 @@ public abstract static class length extends ExternalPropertyNode {
     @Specialization
     @TruffleBoundary
     protected long eval(VmMapping self) {
-      MutableLong count = new MutableLong(0);
-      self.iterateMemberValues(
-          (key, member, value) -> {
+      var count = new MutableLong(0);
+      var visited = new HashSet<>();
+      self.iterateMembers(
+          (key, member) -> {
+            var alreadyVisited = !visited.add(key);
+            // important to record hidden member as visited before skipping it
+            // because any overriding member won't carry a `hidden` identifier
+            if (alreadyVisited || member.isLocalOrExternalOrHidden()) return true;
```

Comment:
This implementation is changed because `iterateMemberValues` might cause a `VmMapping`/`VmListing` to throw, because `VmListingOrMapping#getCachedValue()` can possibly perform an additional typecheck. `Mapping#length` should not need to execute any members, so, it should not need to run typechecks.